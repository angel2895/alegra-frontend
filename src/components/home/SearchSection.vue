<template>
  <section class="search">
    <div class="search-layer-z1">
      <image-search></image-search>
    </div>
    <div class="search-layer-z2">
      <div class="search-layer-z2-container">
        <title-search></title-search>
        <box-search></box-search>
      </div>
    </div>
  </section>
</template>

<script lang="ts">
import { Options, Vue } from "vue-class-component";

import ImageSearch from "@/components/home/search-section/layer-z1/ImageSearch.vue";
import TitleSearch from "@/components/home/search-section/layer-z2/TitleSearch.vue";
import BoxSearch from "@/components/home/search-section/layer-z2/BoxSearch.vue";

@Options({
  components: { ImageSearch, TitleSearch, BoxSearch },
})
export default class SearchSection extends Vue {}
</script>

<style scoped>
.search {
  width: 100%;
  height: 300px;
  padding: 0px;
}

.search-layer-z1,
.search-layer-z2 {
  position: absolute;
  top: 0px;
  width: 100%;
  height: 300px;

  display: flex;
}

.search-layer-z1 {
  z-index: 1;

  align-items: flex-end;
  background-color: var(--background-image);
}

.search-layer-z2 {
  z-index: 2;

  align-items: center;
  justify-content: center;
}

.search-layer-z2-container {
  color: white;
}
</style>
