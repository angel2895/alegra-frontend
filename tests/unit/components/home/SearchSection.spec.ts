import { shallowMount } from "@vue/test-utils";
import SearchSection from "@/components/home/SearchSection.vue";

describe("SearchSection.vue", () => {
  it("Match with snapshot", () => {
    const wrapper = shallowMount(SearchSection);
    expect(wrapper.html()).toMatchSnapshot();
  });
  it("ImageSearch, TitleSearch and BoxSearch are mounted in SearchSection", () => {
    const wrapper = shallowMount(SearchSection);
    expect(wrapper.find("image-search-stub").exists()).toBeTruthy();
    expect(wrapper.find("title-search-stub").exists()).toBeTruthy();
    expect(wrapper.find("box-search-stub").exists()).toBeTruthy();
  });
});
