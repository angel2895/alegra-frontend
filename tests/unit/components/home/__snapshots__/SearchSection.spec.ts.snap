// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchSection.vue Match with snapshot 1`] = `
<section class="search">
  <div class="search-layer-z1">
    <image-search-stub></image-search-stub>
  </div>
  <div class="search-layer-z2">
    <div class="search-layer-z2-container">
      <title-search-stub></title-search-stub>
      <box-search-stub></box-search-stub>
    </div>
  </div>
</section>
`;
